shard-operator-parent/shard-operator-v2/src/main/java/com/redhat/service/smartevents/shard/operator/v2/controllers/ManagedBridgeController.java

```diff
@@ -125,6 +131,22 @@ public UpdateControl<ManagedBridge> reconcile(ManagedBridge managedBridge, Conte
             status.markConditionTrue(DP_NETWORK_RESOURCE_READY_NAME);
         }
 
+        // Create/Update KafkaSource
```

Comment:
I'm not a fan of the `Condition`'s being added to/removed from in the `Reconciler`.

Can we not determine if a Kafka Source is needed from the incoming payload (in `ManagedBridgeService.createManagedBridge(..)` and initialise the set of `Conditions` correctly there?